### PR TITLE
Toolbar and Navigation Drawer

### PR DESCRIPTION
**Pull request by killbotXD**
added toolbar and navigation drawer as a component
Screenshot from 2019-07-12 18-27-38

### CODE REVIEW

File: src/components/Toolbar.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  v-toolbar(clipped-left fixed app)
+    v-toolbar-side-icon(@click.stop="drawer= !drawer")
+    v-toolbar-title {{ title }}
+    v-spacer
+    v-toolbar-items(hidden-sm-and-down)
+      v-btn(flat)
+        v-icon mdi-forum
+        span Forum
+      v-btn(flat href="../konnekt")
+        v-icon mdi-web
```
Comment (ajatprabha) [inline src/components/Toolbar.vue]: Change `v-icon mdi-web` to `v-icon(left) mdi-web`

File: src/components/Toolbar.vue [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        span Konnekt
+      v-menu
+        template(v-slot:activator="{ on }")
+          v-btn(flat v-on="on")
+            v-icon mdi-account
```
Comment (ajatprabha) [inline src/components/Toolbar.vue]: Change `v-icon mdi-account` to `v-icon(left) mdi-account`

File: src/components/Toolbar.vue [hunk 3]
```diff
@@ -0,0 +1,76 @@
+<template lang="pug" xmlns:v-slot="http://www.w3.org/1999/XSL/Transform">
+ div
+  v-toolbar(clipped-left fixed app)
+    v-toolbar-side-icon(@click.stop="drawer= !drawer")
+    v-toolbar-title {{ title }}
+    v-spacer
+    v-toolbar-items(hidden-sm-and-down)
+      v-btn(flat)
+        v-icon mdi-forum
```
Comment (ajatprabha) [inline src/components/Toolbar.vue]: Change `v-icon mdi-forum` to `v-icon(left) mdi-forum`

File: src/components/Toolbar.vue (hunk 1 above)
Comment (killbotXD) [reply src/components/Toolbar.vue]: done

File: src/components/Toolbar.vue (hunk 2 above)
Comment (killbotXD) [reply src/components/Toolbar.vue]: changed as requested

File: src/components/Toolbar.vue (hunk 3 above)
Comment (killbotXD) [reply src/components/Toolbar.vue]: done

File: src/components/Toolbar.vue [hunk 4]
```diff
@@ -6,21 +6,21 @@
...
         template(v-slot:activator="{ on }")
           v-btn(flat v-on="on")
-            v-icon mdi-account
-            span Account
-            v-icon mdi-menu-down
+            v-icon(left) mdi-account
+            span(v-if="$vuetify.breakpoint.mdAndUp") Account
+            v-icon(v-if="$vuetify.breakpoint.mdAndUp") mdi-menu-down
         v-list
           v-list-tile(v-for="item in AccountItems")
```
Comment (ajatprabha) [inline src/components/Toolbar.vue]: This should also have a `:key="i"`, use `v-list-tile(v-for="(item, i) in accountItems" :key="i")`

File: src/views/Konnekt.vue [hunk 5]
```diff
@@ -1,24 +1,30 @@
...
 import Toolbar from "../components/Toolbar";
 
 export default {
   name: "Konnekt",
   components: {
-    Toolbar,
-    Home
+    Toolbar
+  },
+  computed: {
```
Comment (ajatprabha) [inline src/views/Konnekt.vue]: Computed property doesn't work properly. I tried another way:
```js
  data: () => ({ isHome: true }),
  mounted() {
    this.updateBackground();
  },
  methods: {
    updateBackground() { this.isHome = this.$router.currentRoute.path.toString() === "/konnekt"; }
  },
  watch: {
    `$route` (to, from) {
      this.updateBackground();
    }
  }
```

File: src/components/KonnektSearch.vue [hunk 6]
```diff
@@ -57,6 +59,7 @@ class Person {
 }
 export default {
```
Comment (ajatprabha) [inline src/components/KonnektSearch.vue]: Line 15 is causing some issues, change that to `v-container(class="v-responsive__sizer v-responsive__content")` if it is required. Although can you explain why this is needed?

File: src/router.js [hunk 7]
```diff
@@ -29,11 +29,14 @@ export default new Router({
     {
       path: "/konnekt",
       name: "konnekt",
-      component: () => import("./views/Konnekt")
-    },
-    {
-      path: "/konnekt/search",
-      component: () => import("./components/KonnektSearch")
+      component: () => import("./views/Konnekt"),
```
Comment (ajatprabha) [inline src/router.js]: Change these to
```js
    {
      path: "/konnekt",
      component: Konnekt,
      children: [
        { name: "konnekt-home", path: "", component: KonnektHome },
        {
          name: "konnekt-search",
          path: "search",
          component: KonnektSearch
        }
      ]
    }
```
and use import statements at the top of the file, I think chunking all the components is not required.

File: src/components/Toolbar.vue (hunk 4 above)
Comment (ajatprabha) [reply src/components/Toolbar.vue]: Also, instance variables should start with lower case `accountItems`

File: src/components/KonnektSearch.vue (hunk 6 above)
Comment (killbotXD) [reply src/components/KonnektSearch.vue]: no it's not needed actually. I was trying to explore v-responsive.

File: src/components/Toolbar.vue (hunk 4 above)
Comment (killbotXD) [reply src/components/Toolbar.vue]: Okay I will make sure that doesn't happen next time.

File: src/router.js (hunk 7 above)
Comment (killbotXD) [reply src/router.js]: done

File: src/components/KonnektSearch.vue (hunk 6 above)
Comment (killbotXD) [reply src/components/KonnektSearch.vue]: removed them

File: src/views/Konnekt.vue (hunk 5 above)
Comment (killbotXD) [reply src/views/Konnekt.vue]: okay Lets use this then.

Comment (killbotXD) [reply src/views/Konnekt.vue]: done

Comment (killbotXD) [reply src/views/Konnekt.vue]: there's a warning for watch property it says that the parameters `to` and `from` are not used

**Pull request merged**